From 5d2f400cef10bbe7e5c6eee77c7368eb3500e164 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Fri, 27 Nov 2015 00:03:20 -0500
Subject: [PATCH 0532/1110] drm/amd/dal: Add delay for MST after LT

This gives the receiver some time to establish the link
and seems to have a positive impact on ACT handled.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index ec85e87..0e97180 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -1001,6 +1001,7 @@ enum dc_status core_link_enable(struct core_stream *stream)
 		break;
 	case SIGNAL_TYPE_DISPLAY_PORT_MST:
 		status = enable_link_dp_mst(stream);
+		dc_service_sleep_in_milliseconds(stream->ctx, 200);
 		break;
 	case SIGNAL_TYPE_DVI_SINGLE_LINK:
 	case SIGNAL_TYPE_DVI_DUAL_LINK:
-- 
2.7.4

